packages/squiggle-lang/src/utility/DateTime.ts

```diff
@@ -80,11 +89,17 @@ export const DateModule = {
   },
   makeFromYear(year: number): result<Date, string> {
     const floor = Math.floor(year);
-    return Result.fmap(DateModule.makeWithYearInt(floor), (earlyDate) => {
+    return Result.fmap(date.makeWithYearInt(floor), (earlyDate) => {
       const diff = year - floor;
-      return DateModule.addDuration(earlyDate, diff * Duration.year);
+      return date.addDuration(earlyDate, diff * DurationUnits.Year);
     });
   },
+  makeFromString(str: string): result<Date, string> {
+    const parsedDate = new Date(str);
+    if (isNaN(parsedDate.getTime())) {
```

Comment:
maybe I should apply this elsewhere? It is annoying though.

Reply:
Maybe ``    if (parsedDate instanceof Date && isFinite(parsedDate.getTime())) {``?

``isNaN(parsedDate)`` doesn't typecheck in TS for me.